admin-tools/lib/src/main/scala/com/gu/mediaservice/ImagesBatchProjection.scala

```diff
@@ -0,0 +1,38 @@
+package com.gu.mediaservice
+
+import com.gu.mediaservice.lib.config.{ServiceHosts, Services}
+import play.api.libs.json.Json
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object ImagesBatchProjection {
+  def apply(apiKey: String, domainRoot: String): ImagesBatchProjection =
+    new ImagesBatchProjection(apiKey, domainRoot)
+}
+
+case class ImageIdMaybeBlobEntry(id: String, blob: Option[String])
+
+case class ImageIdBlobEntry(id: String, blob: String)
```

Comment:
Why do we need this and `ImageIdMaybeBlobEntry`?

Reply:
to be able to track which images were successfully projected

ImageIdMaybeBlobEntry will give you id and maybe something
and then it will be mapped to lis of ImageIdBlobEntry (success) and list of String (failed projection id) in `BatchIndexHandler.partitionToSuccessAndNotFound `function

such that we can record state in Dynamo
1 -> in progress
2 -> not found
3 -> all done

Reply:
i didn't want to use pure tuples